agent/lib/kubernetes.js

```diff
@@ -70,8 +70,15 @@ function get_options(options, path) {
 function get_path(base, resource, options) {
     var namespace = options.namespace || process.env.KUBERNETES_NAMESPACE || read('/var/run/secrets/kubernetes.io/serviceaccount/namespace');
     var path = base + namespace + '/' + resource;
+    var queryParams = {};
     if (options.selector) {
-        path += '?' + querystring.stringify({'labelSelector':options.selector});
+        queryParams.labelSelector = options.selector;
+    }
+    if (options.requestTimeout) {
+        queryParams.timeoutSeconds = options.requestTimeout;
+    }
+    if (Object.keys(queryParams).length) {
```

Comment:
I know it works, but maybe use length > 0 just to help the reader :)

Reply:
Fix commited.